docs/structure-of-a-contract.rst

```diff
@@ -31,6 +31,25 @@ See the :ref:`types` section for valid state variable types and
 :ref:`visibility-and-getters` for possible choices for
 visibility.
 
+You can initialize a state variable inline for all types (even for structs). But for arrays you must declare them as static memory arrays. For example::
```

Comment:
We should explain that inline initialization happens before the constructor is run.